node/store.go

```diff
@@ -66,31 +66,108 @@ func NewLevelDBStore(path string, logger logging.Logger, metrics *Metrics, block
 // The function returns the number of batches deleted and the status of deletion. Note that the
 // number of batches deleted can be positive even if the status is error (e.g. the error happened
 // after it had successfully deleted some batches).
-func (s *Store) DeleteExpiredEntries(currentTimeUnixSec int64, timeLimitSec uint64) (int, error) {
+func (s *Store) DeleteExpiredEntries(currentTimeUnixSec int64, timeLimitSec uint64) (numBatchesDeleted int, numBlobsDeleted int, err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeLimitSec)*time.Second)
 	defer cancel()
 
-	numBatchesDeleted := 0
+	numBatchesDeleted = 0
+	numBlobsDeleted = 0
 	for {
 		select {
 		case <-ctx.Done():
-			return numBatchesDeleted, ctx.Err()
+			return numBatchesDeleted, numBlobsDeleted, ctx.Err()
 		default:
-			numDeleted, err := s.deleteNBatches(currentTimeUnixSec, numBatchesToDeleteAtomically)
+			numDeleted, err := s.deleteExpiredBlobs(currentTimeUnixSec, numBatchesToDeleteAtomically)
 			if err != nil {
-				return numBatchesDeleted, err
+				return numBatchesDeleted, numBlobsDeleted, err
+			}
+			numBlobsDeleted += numDeleted
+
+			numDeleted, err = s.deleteNBatches(currentTimeUnixSec, numBatchesToDeleteAtomically)
+			if err != nil {
+				return numBatchesDeleted, numBlobsDeleted, err
 			}
 			// When there is no error and we didn't delete any batch, it means we have
 			// no obsolete batches to delete, so we can return.
 			if numDeleted == 0 {
```

Comment:
This seems like it will return if there are no more batches to delete even if there are still blobs to delete.

Reply:
Thanks for catching this!